packages/paste-style-props/src/types/flexbox.ts

```diff
@@ -11,74 +11,100 @@ import {ResponsiveValue, TLengthStyledSystem} from 'styled-system';
  * along the main-axis of a flex container, and the inline axis of a grid container.
  *
  * [MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content)
  */
-export type JustifyContent = ResponsiveValue<
-  CSS.Globals | CSS.ContentDistribution | CSS.ContentPosition | 'left' | 'normal' | 'right'
->;
+export type JustifyContentOptions =
+  | CSS.Globals
+  | CSS.ContentDistribution
+  | CSS.ContentPosition
+  | 'left'
+  | 'normal'
+  | 'right';
+export type JustifyContent = ResponsiveValue<JustifyContentOptions>;
 
 /**
  * The flex-wrap CSS property sets whether flex items are forced onto one line or can wrap onto multiple lines.
  * If wrapping is allowed, it sets the direction that lines are stacked.
  *
  * [MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-wrap)
  */
-export type FlexWrap = ResponsiveValue<CSS.FlexWrapProperty>;
+export type FlexWrapOptions = CSS.FlexWrapProperty;
+export type FlexWrap = ResponsiveValue<FlexWrapOptions>;
 
 // TODO: The FlexBasisValue currently really only exists for documentation
 //       purposes, because flex-basis also accepts `Nem` and `Npx` strings.
 //       Not sure there’s a way to still have the union values show up as
 //       auto-completion results.
-export type FlexBasis = ResponsiveValue<CSS.FlexBasisProperty<TLengthStyledSystem>>;
+export type FlexBasisOptions = CSS.FlexBasisProperty<TLengthStyledSystem>;
+export type FlexBasis = ResponsiveValue<FlexBasisOptions>;
 
 /**
  * The flex-direction CSS property specifies how flex items are placed in the flex container defining the main
  * axis and the direction (normal or reversed).
  *
  * [MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction)
  */
-export type FlexDirection = ResponsiveValue<CSS.FlexDirectionProperty>;
+export type FlexDirectionOptions = CSS.FlexDirectionProperty;
+export type FlexDirection = ResponsiveValue<FlexDirectionOptions>;
 
 /**
  * The flex CSS property specifies how a flex item will grow or shrink so as to fit the space available in
  * its flex container. This is a shorthand property that sets flex-grow, flex-shrink, and flex-basis.
  *
  * [MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/flex)
  */
-export type Flex = ResponsiveValue<CSS.FlexProperty<TLengthStyledSystem>>;
+export type FlexOptions = CSS.FlexProperty<TLengthStyledSystem>;
+export type Flex = ResponsiveValue<FlexOptions>;
 
 /**
  * The CSS justify-self property set the way a box is justified inside its alignment container along
  * the appropriate axis.
  *
  * [MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/justify-self)
  */
+export type JustifySelfOptions =
+  | CSS.Globals
+  | CSS.SelfPosition
+  | 'auto'
+  | 'baseline'
+  | 'left'
+  | 'normal'
+  | 'right'
+  | 'stretch'
+  | string;
 export type JustifySelf =
   | ResponsiveValue<CSS.Globals | CSS.SelfPosition | 'auto' | 'baseline' | 'left' | 'normal' | 'right' | 'stretch'>
```

Comment:
Should this line be `ResponsiveValue<JustifySelfOptions>`?

Reply:
/shrug
Spoke in person